### PR TITLE
Opentelemetry from TP to full supported

### PR DESCRIPTION
**Pull request by jcarranzan**
### Links

JIRA:
https://issues.redhat.com/browse/QUARKUS-2391 
Quarkus documentation:
[Opentelemetry-quickstart code](https://github.com/quarkusio/quarkus-quickstarts/tree/main/opentelemetry-quickstart)
### Reminder for considerable topics

 - [x] Make sure you have considered the following areas when preparing the test plan:

   - Logging
   - Tracing
   - Metrics
   - Security
   - OpenAPI
   - Data sources
   - Frontend
   - Qute

### CODE REVIEW

File: QUARKUS-2391.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+#### Test scenarios
+- Disable Opentelemetry extensions to ensure the application behaves as expected without tracing enabled.
+- Testing scenario with additional configuration options such as Id Generator, Propagators, Resource, Sampler configuration.
+- Test with other exporters such a Zipkin.
+- Integration tests to ensure proper tracing is enabled and functioning correctly in other core Quarkus extensions such as ```quarkus-smallrye-reactive-messaging-amqp```.
+
+### Impact on testsuites and testing automation:
+New tests will be added inside monitoring-opentelemetry modules and messaging-amqp-reactive.
+And ensure that test cases are running in both JVM and OpenShift environments.
```
Comment (fedinskiy) [inline QUARKUS-2391.md:29]: We usually have matrix of 4 environments: JVM/Native and Baremetal/Openshift. Which are we going to use there?

File: QUARKUS-2391.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+The traces are directly sent to the Jaeger collector, and they also test Quarkus gRPC communication between services and SSE (Server-Sent Events).
+
+On the other hand, also other modules use ```quarkus-opentelemetry``` dependency:
+- javaee-like-getting-started
+- build-time-analytics
+- http/graphql-telemetry
+- http/vertx-web-client
+- sql-db/narayana-transactions (where is tested JDBC Opentelemetry instrumentation)
+
+#### Test scenarios
```
Comment (fedinskiy) [inline QUARKUS-2391.md:21]: Are these scenarios already exist or do we need to add them?

File: QUARKUS-2391.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+- Disable Opentelemetry extensions to ensure the application behaves as expected without tracing enabled.
+- Testing scenario with additional configuration options such as Id Generator, Propagators, Resource, Sampler configuration.
+- Test with other exporters such a Zipkin.
+- Integration tests to ensure proper tracing is enabled and functioning correctly in other core Quarkus extensions such as ```quarkus-smallrye-reactive-messaging-amqp```.
```
Comment (fedinskiy) [inline QUARKUS-2391.md:25]: Also, we recently dropped opentracing, so we can try to move grpc coverage to opentelemetry as well as well: https://github.com/quarkus-qe/quarkus-test-plans/blob/main/QUARKUS-3660.md?plain=1#L17

**Review by fedinskiy — CHANGES_REQUESTED** [review]

File: QUARKUS-2391.md (hunk 2 above)
Comment (jcarranzan) [reply QUARKUS-2391.md:21]: No, they are not, we need to add them.

File: QUARKUS-2391.md (hunk 3 above)
Comment (jcarranzan) [reply QUARKUS-2391.md:25]: Ok, as far as I know, in the current monitoring-opentelemetry modules tests we already have some gRPC coverage integrated here: 
https://github.com/quarkus-qe/quarkus-test-suite/blob/85a50304c23d39a264eb0530872c57560ceb2c02/monitoring/opentelemetry/pom.xml#L36

https://github.com/quarkus-qe/quarkus-test-suite/blob/85a50304c23d39a264eb0530872c57560ceb2c02/monitoring/opentelemetry-reactive/pom.xml#L32

But not a full module dedicated as you mentioned in that test plan: 
https://github.com/quarkus-qe/quarkus-test-suite/tree/main/monitoring/opentracing-reactive-grpc

So, it's a good idea to include it, thanks.

File: QUARKUS-2391.md (hunk 1 above)
Comment (jcarranzan) [reply QUARKUS-2391.md:29]: I wrote down JVM (I think by default all the tests are running in bare metal/JVM mode)  and Openshift (keeping the monitoring Opentelemetry testing strategy). Not sure, if in this case, we can provide some value also to include Native execution.

Comment (fedinskiy) [reply QUARKUS-2391.md]: When, please, write this more clearly, "running on JVM in both baremetal and OpenShift environmeents"

File: QUARKUS-2391.md (hunk 3 above)
Comment (fedinskiy) [reply QUARKUS-2391.md]: If we have the same coverage in `opentelementry-reactive`, we can left it here and just drop opentracing. The only important thing is to have coverage for grpc+opentelemetry, it doesn't matter, in which module.

**Review by fedinskiy — CHANGES_REQUESTED** [review]
see above

File: QUARKUS-2391.md (hunk 1 above)
Comment (jcarranzan) [reply QUARKUS-2391.md]: I apologize for any confusion, I've corrected the sentence for clarification, thanks @fedinskiy .

**Review by fedinskiy — APPROVED** [review]

**Pull request merged**